server/src/main/java/org/elasticsearch/search/fetch/FetchSubPhase.java

```diff
@@ -36,19 +35,21 @@
 public interface FetchSubPhase {
 
     class HitContext {
-        private SearchHit hit;
-        private IndexSearcher searcher;
-        private LeafReaderContext readerContext;
-        private int docId;
+        private final SearchHit hit;
+        private final IndexSearcher searcher;
+        private final LeafReaderContext readerContext;
+        private final int docId;
         private final SourceLookup sourceLookup = new SourceLookup();
-        private Map<String, Object> cache;
+        private final Map<String, Object> cache;
```

Comment:
Some of the highlighter impls use it; I agree we should be able to remove it but I think it will need some changes to the Highlighter API (HighlightPhase can have state, but highlighters themselves don't) so I'll add a TODO and we can do it in a followup.

Reply:
yes I know, what I was trying to understand is whether that will be an issue to change the behavior of this map. If I understand correctly, we don't clear the map after each document, so I have no idea if this is acceptable for the usage that we have.

Reply:
OIC, yes, this is actually preserving previous behaviour - we used to call `HitContext.reset()` which would change alter everything *except* the cache, so that things put there would be re-used across different hits.